src/main/java/com/blibli/future/config/WebSecurityConfiguration.java

```diff
@@ -0,0 +1,58 @@
+package com.blibli.future.config;
+
+
+import com.blibli.future.security.JwtAuthenticationEntryPoint;
+import com.blibli.future.security.JwtAuthenticationFilter;
+import com.blibli.future.security.JwtAuthenticationProvider;
+import com.blibli.future.security.JwtAuthenticationSuccessHandler;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+public class WebSecurityConfiguration extends WebSecurityConfigurerAdapter {
+    @Autowired
+    private JwtAuthenticationProvider authenticationProvider;
+    @Autowired
+    private JwtAuthenticationEntryPoint authenticationEntryPoint;
+
+    public JwtAuthenticationFilter authenticationFilter() throws Exception {
+        JwtAuthenticationFilter filter = new JwtAuthenticationFilter();
+        filter.setAuthenticationSuccessHandler(new JwtAuthenticationSuccessHandler());
+        filter.setAuthenticationManager(this.authenticationManager());
+        return filter;
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.csrf().disable()
+                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and()
+                .authorizeRequests()
+                .antMatchers("/shift").hasRole("ADMIN")
```

Comment:
Well, this one is just an example on how to do it, since I'm not too sure which module has which role. 
I hope you guys will help me finish the overall authorization path if possible 😀